server/src/main/java/com/cloud/configuration/ConfigurationManagerImpl.java

```diff
@@ -4067,21 +4166,21 @@ public Vlan createVlanAndPublicIpRange(final CreateVlanIpRangeCmd cmd) throws In
                 zoneId = network.getDataCenterId();
                 physicalNetworkId = network.getPhysicalNetworkId();
             }
-        } else if (ipv6) {
+        }/* else if (ipv6) {
             throw new InvalidParameterValueException("Only support IPv6 on extending existed network");
-        }
+        }*/
 
         // Verify that zone exists
         final DataCenterVO zone = _zoneDao.findById(zoneId);
         if (zone == null) {
             throw new InvalidParameterValueException("Unable to find zone by id " + zoneId);
         }
 
-        if (ipv6) {
-            if (network.getGuestType() != GuestType.Shared || zone.isSecurityGroupEnabled()) {
-                throw new InvalidParameterValueException("Only support IPv6 on extending existed share network without SG");
-            }
-        }
+//        if (ipv6) {
```

Comment:
@weizhouapache `Ipv6SubnetCapacityThreshold` holds the threshold value for IPv6 subnets and this code adds public vlan (IPv6) range. Can you please elaborate the check that should be added?

Reply:
@shwstppr 
oh ,sorry my mistake, it should be global setting `Ipv6NetworkOfferingCreationEnabled`.
ps: would it better to rename the setting to `Ipv6NetworkEnabled` ?

Reply:
@weizhouapache Offering creation was guarded behind global setting with suggestion from @DaanHoogland so the IPv6 offering doesn't get presented to normal user when the offering is created by mistake. Adding IPv6 range is again an admin only functionality and it will work only when admin will provide valid IPv6 range/prefix. Does it still need to be guarded with global setting?
Global setting can be renamed but `Ipv6NetworkEnabled` might be confusing with IPv6 shared networks functionality.